src/Compiler/HeapReporter.cpp

```diff
@@ -48,15 +39,23 @@ void logMessage(StringRef logFilename, StringRef msg,
 #endif
 } // namespace
 
-HeapReporter::HeapReporter(
-    std::string logFilename, StringRef beforePasses, StringRef afterPasses)
+HeapReporter::HeapReporter(std::string logFilename,
+    std::vector<std::string> beforePasses, std::vector<std::string> afterPasses)
     : logFilename(logFilename) {
-  splitToSet(beforePasses, this->beforePassesSet);
-  splitToSet(afterPasses, this->afterPassesSet);
+  beforePassesSet = llvm::StringSet<>(beforePasses);
+  afterPassesSet = llvm::StringSet<>(afterPasses);
```

Comment:
can you set before/afterPassesSet in the initialization list?
```
  : logFilename(logFilename), beforePassesSet(beforePasses), afterPassesSet(afterPasses) {}
```

Reply:
Good catch